editor/effects.js

```diff
@@ -20,9 +20,30 @@ import {
 	getCurrentPostType,
 	getBlocks,
 	getPostEdits,
+	getDocumentTitle,
 } from './selectors';
 
+let originalDocumentTitle; // @todo Obtain document.title template from PHP export?
+function populateDocumentTitle( title ) {
+	if ( ! originalDocumentTitle ) {
+		originalDocumentTitle = document.title;
+	}
+	document.title = title + ' | ' + originalDocumentTitle;
+}
+
 export default {
+	RESET_POST( action, store ) {
+		setTimeout( () => { // Next-tick to ensure action has been applied to state in store .
```

Comment:
An alternative here is to not treat these as effects of the action, but rather handled by logic which has subscribed to store updates (e.g. a `connect`ed component, or a `store.subscribe` callback).

This is the [approach we had taken in the Calypso editor](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/editor-document-head/index.jsx). Basically: Instead of binding to the specific actions that we think would cause a change to the document title selector value, we instead re-run the selector any time any dispatch occurs, and if the result is different, apply the new title value.